substrate/validator-sets/pallet/src/lib.rs

```diff
@@ -127,6 +187,227 @@ pub mod pallet {
 
       Ok(())
     }
+
+    /// Will set the given account to validator list if the allocated bond is enough.
+    pub fn add_participant(
+      account: T::AccountId,
+      amount: Amount,
+      network: NetworkId,
+    ) -> DispatchResult {
+      let key = ValidatorSet { session: Self::current_session(), network };
+      if amount < Self::validator_set(key).unwrap().bond {
+        Err(Error::<T>::InSufficientBond)?;
+      }
+
+      // Since this participant will be included after the next session, save it for the set
+      // where it will be active.
+      let participant_set =
+        ValidatorSet { session: Session(Self::current_session().0 + 2), network };
```

Comment:
yea we can set the rotation time to 3.5 days, but it might still be better for other networks to follow the serai net? that would purify the design, since all the commands would be coming from the session pallet. Otherwise  we are kinda looking at  implementing our own session pallet for other networks. Which if we were gonna do that, we can also use it for the serai net as well. But yea migration taking 6 hrs every 3.5 days is also a problem in that case.